darwin/dataset/remote_dataset.py

```diff
@@ -78,61 +83,79 @@ def __init__(
 
     def push(
         self,
-        files_to_upload: List[str],
+        files_to_upload: Optional[List[Union[str, Path, LocalFile]]],
```

Comment:
This is debatable, but I would force all arguments but `files_to_upload` to be named.

Reply:
Given that push is a public function part of darwin's public API, wouldn't such a change possibly break clients that are already invoking it with positional parameters only?

Reply:
Yes, that's why I wanted to bring this conversation up. We are changing the upload API entirely, and this belongs to a `0.6` release, so I'd be fine with that, but let's hear what @simedw or @myFree think

Reply:
If we are changing it, now is the time.
Let's do it.